cg/models/demultiplex/flow_cell.py

```diff
@@ -77,7 +77,26 @@ def full_name(self) -> str:
     @property
     def sample_sheet_path(self) -> Path:
         """Return sample sheet path."""
-        return Path(self.path, DemultiplexingDirsAndFiles.SAMPLE_SHEET_FILE_NAME)
+        root_sample_sheet_path: Path = Path(
+            self.path, DemultiplexingDirsAndFiles.SAMPLE_SHEET_FILE_NAME
+        )
+
+        unaligned_sample_sheet_path: Path = Path(
+            self.path,
+            DemultiplexingDirsAndFiles.UNALIGNED_DIR_NAME,
+            DemultiplexingDirsAndFiles.SAMPLE_SHEET_FILE_NAME,
+        )
+
+        sample_sheet_paths: List[Path] = [
+            root_sample_sheet_path,
+            unaligned_sample_sheet_path,
+        ]
+
+        for sample_sheet in sample_sheet_paths:
+            if sample_sheet.exists():
+                return sample_sheet
+
+        return root_sample_sheet_path
```

Comment:
A bit confused here. Given that there is no priority given (although I guess the two paths are mutually exclusive) and given that there is no error raised if neither exists, it seems to me as if unaligned_sample_sheet_path is returned if it exists while root_sample_sheet_path doesn't, and in. all other instances root_sample_sheet_path is returned? Should this be the case?

Reply:
I agree that it is confusing.

This property is sometimes used in a context when the sample sheet has not been created yet. So unless a sample sheet has been created previously, we return the default root path to where it should be created.

Reply:
Okay, but would it not make sense to have it as 
"return unaligned if (unaligned.exists and not root.exists) else root"?

I feel that the list where the behaviour depends on the order implicitly confuses me a bit.

Reply:
I think we should refactor it in a separate PR, I created this issue before to track it https://github.com/Clinical-Genomics/cg/issues/2223

Reply:
Looks nice!

Reply:
It is not really a big thing, I just think the for-loop is a bit confusing given that the result essentially is "return unaligned if unaligned.exists else root". The way it is written, it would be more intuitive if we raise an error given that neither root nor unaligned exist. Probably won't affect functionality, though, so feel free to ignore if you see no benefit from a change.

Reply:
I'm in favor of the solution @islean presented

Reply:
Yeah, I agree and will remove the for loop. I'm not going to raise an error though because that would be a breaking change and is out of scope for this PR. Also, @karlnyr says we cannot assume that a sample sheet is present in the demultiplexed flow cell directory

Reply:
Which is an issue I think will have to be addressed in a separate PR as well. We should add a step in the new demux flow where we copy the sample sheet from the run directory into the demux directory

Reply:
I restored the original version, I don't want to touch anything outside the new post processing flow in this PR. And the change I made here does not really make sense, we just need to make sure that the new post processing flow transfers the sample sheet to the demux dir. That is a separate PR though.